src/Traces.Web/Services/ApaleoOneService.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Optional;
+using Traces.Common.Constants;
+using Traces.Common.Exceptions;
+using Traces.Web.Models;
+
+namespace Traces.Web.Services
+{
+    public class ApaleoOneService : IApaleoOneService
+    {
+        private readonly IJSRuntime _jsRuntime;
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver()
+        };
+
+        public ApaleoOneService(IJSRuntime runtime)
+        {
+            _jsRuntime = runtime;
+        }
+
+        public async Task<ResultModel<bool>> NavigateToReservationAsync(TraceItemModel traceItemModel)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(traceItemModel.ReservationId) ||
```

Comment:
```suggestion
                if (string.IsNullOrWhiteSpace(traceItemModel.ReservationId) ||
```
Also for the one below